core/chains/legacyevm/chain.go

```diff
@@ -475,13 +475,19 @@ func newEthClientFromCfg(cfg evmconfig.NodePool, noNewHeadsThreshold time.Durati
 	for i, node := range nodes {
 		if node.SendOnly != nil && *node.SendOnly {
 			name := fmt.Sprintf("eth-sendonly-rpc-%d", i)
+			if node.Name != nil && *node.Name != "" {
+				name = *node.Name
+			}
 			rpc := evmclient.NewRPCClient(lggr, empty, (*url.URL)(node.HTTPURL), name, int32(i), chainID,
 				commonclient.Secondary)
```

Comment:
It is safe to pass the `nodeName` directly. We validate that all nodes are named. Notice that we already pas it without nil check on the line below. 
```suggestion
			rpc := evmclient.NewRPCClient(lggr, empty, (*url.URL)(node.HTTPURL), *node.Name, int32(i), chainID,
				commonclient.Secondary)
```